scripts/tutorials/generate_database.py

```diff
@@ -38,14 +31,48 @@
         # Database already empty or doesn't yet exist
         pass
 
-create_database(connection_string)
+    # Proceed to load the database
+    load_database(connection_string)
+
+
+def load_sqlite():
+    # First, remove the existing database in order to recreate it from the schema
+    # If the schema has not changed, this part can be skipped
+    if os.path.exists(DB_NAME):
+        os.remove(DB_NAME)
+    connection_string = 'sqlite:///' + DB_NAME
+
+    # Proceed to load the database
+    load_database(connection_string)
+
+
+def load_database(connection_string):
+    # Create and load the database
+    create_database(connection_string)
+
+    # Now that the database is created, connect to it and load up the JSON data
+    db = Database(connection_string, reference_tables=REFERENCE_TABLES)
+    db.load_database(DB_PATH, verbose=False)
+
+    print('New database generated.')
 
-# Now that the database is created, connect to it and load up the JSON data
-db = Database(connection_string, reference_tables=REFERENCE_TABLES)
-db.load_database(DB_PATH, verbose=False)
+    # Close all connections
+    db.session.close()
+    db.engine.dispose()
 
-print('New database generated.')
 
-# Close all connections
-db.session.close()
-db.engine.dispose()
+if __name__ == '__main__':
+    if len(sys.argv) > 1:
+        architecture = sys.argv[1].lower()
+        if architecture == 'postgres':
+            if len(sys.argv) > 2:
+                connection_string = sys.argv[2]
+            else:
+                connection_string = os.getenv('SIMPLE_DATABASE_URL')
+            load_postgres(connection_string)
+        elif architecture == 'sqlite':
+            load_sqlite()
+        else:
+            print(f'Unrecognized architecture: {architecture}')
+    else:
+        print('Specify database architecture (eg, postgres, sqlite)')
```

Comment:
Not a huge fan of this implementation, you could do this with ```argparse```, no?

Reply:
This was quick and dirty, but looking through `argparse` I like how it works so I'll update it.